src/main/java/it/pagopa/ecommerce/commons/domain/TransactionRefunded.java

```diff
@@ -0,0 +1,65 @@
+package it.pagopa.ecommerce.commons.domain;
+
+import it.pagopa.ecommerce.commons.documents.TransactionRefundedEvent;
+import it.pagopa.ecommerce.commons.domain.pojos.BaseTransaction;
+import it.pagopa.ecommerce.commons.generated.server.model.TransactionStatusDto;
+import lombok.AccessLevel;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import lombok.experimental.FieldDefaults;
+
+/**
+ * <p>
+ * Transaction refunded. This state means that the transaction is not completed
+ * successfully requiring a refund to be performed to the user This is a final
+ * state so any event applied to this class will be discarded
+ * </p>
+ *
+ * @see Transaction
+ * @see BaseTransaction
+ */
+@EqualsAndHashCode(callSuper = true)
+@ToString
+@FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+@Getter
+public final class TransactionRefunded extends BaseTransaction implements Transaction {
```

Comment:
Because we can reach `REFUNDED` from many different states, consider adding a `BaseTransactionRefunded` with a `transactionBeforeRefund` field, like in `BaseTransactionExpired`

Reply:
Fixed in https://github.com/pagopa/pagopa-ecommerce-commons/pull/31/commits/77050e03697fccb38f9939425842963982b17fde